server/src/domain/audit/audit.service.ts

```diff
@@ -40,4 +65,160 @@ export class AuditService {
       ids: audits.map(({ entityId }) => entityId),
     };
   }
+
+  async getChecksums(dto: FileChecksumDto) {
+    const results: FileChecksumResponseDto[] = [];
+    for (const filename of dto.filenames) {
+      if (!StorageCore.isImmichPath(filename)) {
+        throw new BadRequestException(
+          `Could not get the checksum of ${filename} because the file isn't accessible by Immich`,
+        );
+      }
+
+      const checksum = await this.cryptoRepository.hashFile(filename);
+      results.push({ filename, checksum: checksum.toString('base64') });
+    }
+    return results;
+  }
+
+  async fixItems(items: FileReportItemDto[]) {
```

Comment:
We are not perfoming any file moves operation here, how can we fix the orphaned files in the upload folder??

Reply:
When we find out where the actual file is stored, we simply update the database to point to it. We decided to fix it on the database side rather than move the file itself.